go/private/rules/binary.bzl

```diff
@@ -415,60 +416,32 @@ def _go_tool_binary_impl(ctx):
     if sdk.goos == "windows":
         name += ".exe"
 
-    cout = ctx.actions.declare_file(name + ".a")
+    out = ctx.actions.declare_file(name)
     if sdk.goos == "windows":
-        cmd = "@echo off\n {go} tool compile -o {cout} -trimpath=%cd% {srcs}".format(
-            go = sdk.go.path.replace("/", "\\"),
-            cout = cout.path,
-            srcs = " ".join([f.path for f in ctx.files.srcs]),
-        )
-        bat = ctx.actions.declare_file(name + ".bat")
-        ctx.actions.write(
-            output = bat,
-            content = cmd,
-        )
-        ctx.actions.run(
-            executable = bat,
-            inputs = sdk.libs + sdk.headers + sdk.tools + ctx.files.srcs + [sdk.go],
-            outputs = [cout],
-            env = {"GOROOT": sdk.root_file.dirname},  # NOTE(#2005): avoid realpath in sandbox
-            mnemonic = "GoToolchainBinaryCompile",
-        )
+        # TODO(matloob): do the below, but for Windows.
+        pass
     else:
-        cmd = "{go} tool compile -o {cout} -trimpath=$PWD {srcs}".format(
+        # Note: GOPATH is needed for Go 1.16.
+        cmd = "GOCACHE=$(mktemp -d) GOPATH=$(mktemp -d) {go} build -o {out} -trimpath {srcs}".format(
             go = sdk.go.path,
-            cout = cout.path,
+            out = out.path,
             srcs = " ".join([f.path for f in ctx.files.srcs]),
         )
         ctx.actions.run_shell(
             command = cmd,
-            inputs = sdk.libs + sdk.headers + sdk.tools + ctx.files.srcs + [sdk.go],
-            outputs = [cout],
-            env = {"GOROOT": sdk.root_file.dirname},  # NOTE(#2005): avoid realpath in sandbox
+            inputs = sdk.headers + sdk.tools + sdk.srcs + ctx.files.srcs + [sdk.go],  # sdk.libs
+            outputs = [out],
             mnemonic = "GoToolchainBinaryCompile",
         )
 
-    out = ctx.actions.declare_file(name)
-    largs = ctx.actions.args()
-    largs.add_all(["tool", "link"])
-    largs.add("-o", out)
-    largs.add(cout)
-    ctx.actions.run(
-        executable = sdk.go,
-        arguments = [largs],
-        inputs = sdk.libs + sdk.headers + sdk.tools + [cout],
-        outputs = [out],
-        env = {"GOROOT_FINAL": "GOROOT"},  # Suppress go root paths to keep the output reproducible.
-        mnemonic = "GoToolchainBinary",
-    )
-
     return [DefaultInfo(
         files = depset([out]),
         executable = out,
     )]
 
 go_tool_binary = rule(
     implementation = _go_tool_binary_impl,
+    cfg = go_tool_transition,  # For fmeum: Is this correct? do we need go_reset_target?
```

Comment:
`non_go_tool_transition` may perform better here (the bootstrap rules don't use nogo, so the setting shouldn't matter), but better verify that with `cquery`.

Reply:
The recent fix brought it down to just target and exec variants. I don't know where the target variant comes from when building all of rules_go and `cquery` doesn't show me the path. It doesn't seem to be built with a normal `go_binary`, so I'm inclined to say it's not an issue.

I also checked the stdlib rebuilds and they all look reasonable.